server/main_server/src/rodsServer.cpp

```diff
@@ -468,6 +483,347 @@ namespace
+            if (const auto err = get_catalog_service_role(service_role); !err.ok()) {
+                log::error("Could not retrieve server role [error_code={}].", err.code());
+                return;
+            }
+
+            const auto is_provider = (irods::CFG_SERVICE_ROLE_PROVIDER == service_role);
+
+            irods::at_scope_exit disconnect_from_database{[is_provider] {
+                if (is_provider) {
+                    if (const auto ec = chlClose(); ec != 0) {
+                        log::error("Could not disconnect from database [error_code={}].", ec);
+                    }
+                }
+            }};
+
+            if (is_provider) {
+                // Assume this server has database credentials and attempt to connect to the database.
+                if (const auto ec = chlOpen(); ec != 0) {
+                    log::error("Could not connect to database [error_code={}].", ec);
+                    return;
+                }
+            }
+
+            char hostname[HOST_NAME_MAX]{};
+            if (gethostname(hostname, sizeof(hostname)) != 0) {
+                log::error("Failed to retrieve local server's hostname for delay server migration.");
+                return;
+            }
+            log::trace("Local server's hostname = [{}]", hostname);
+
+            std::string leader;
+            std::string successor;
+
+            {
+                ix::client_connection comm;
+
+                auto result = get_grid_configuration_option_value(comm, "delay_server", "leader");
+                if (!result) {
+                    return;
+                }
+
+                leader = std::move(*result);
+                log::trace("Leader's hostname = [{}]", leader);
+
+                result = get_grid_configuration_option_value(comm, "delay_server", "successor");
+                if (!result) {
+                    return;
+                }
+
+                successor = std::move(*result);
+                log::trace("Successor's hostname = [{}]", successor);
+            }
+
+            if (hostname == leader) {
+                log::trace("Local server [{}] is the leader.", hostname);
+
+                if (!successor.empty() && hostname != successor) {
+                    log::trace("New successor detected!");
+
+                    const auto pid = irods::get_delay_server_pid();
+
+                    if (!pid) {
+                        log::trace("Could not get delay server PID.");
+                        return;
+                    }
+
+                    log::trace("Delay server PID = [{}]", *pid);
+                    log::info("Sending shutdown signal to delay server.");
+
+                    // If the delay server is running locally, send SIGTERM to it and wait
+                    // for graceful shutdown. The call to waitpid() will cause the CRON manager
+                    // to stop making process until it returns because it is single threaded.
```

Comment:
stop making process -> stop making processes

Reply:
That is actually supposed to be "progress".

Very good catch. That would've lead to so much confusion down the road.

Reply:
Oh, well, even better then. :)